### PR TITLE
New: Request v2 model fragments

### PR DESCRIPTION
**Pull request by daniellevinson**
⚠️ This is not a breaking change but will not be merged into master

## Description

Model request v2

Not included in this PR:
- Adding metadata that we currently do in transformations
- Multi datasource support (easy to add if we decide to leave it where it is now)
- Model methods used by consumers

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/models/bard-request-v2/request.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+export default Fragment.extend(Validations, {
+  filters: fragmentArray('bard-request-v2/fragments/filter', { defaultValue: [] }),
+  columns: fragmentArray('bard-request-v2/fragments/column', { defaultValue: [] }),
+  table: attr('string'),
+  sort: fragmentArray('bard-request-v2/fragments/sort', { defaultValue: [] }),
+  limit: attr('number'),
+  requestVersion: attr('string', { defaultValue: '2.0' }),
+  dataSource: attr('string', { defaultValue: getDefaultDataSourceName() }),
+  responseFormat: attr('string', { defaultValue: 'json' }),
```
Comment (cythrawll) [inline packages/core/addon/models/bard-request-v2/request.js:55]: Do we need this, is this part of the spec?

Comment (daniellevinson) [reply packages/core/addon/models/bard-request-v2/request.js:55]: I wasn't sure but looks like it can be removed

Comment (jkusa) [reply packages/core/addon/models/bard-request-v2/request.js]: We can remove

File: packages/core/addon/models/bard-request-v2/fragments/filter.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  values: [
+    validator('length', {
+      min: 1,
+      allowNone: false,
+      message() {
+        const { field } = this.model;
+        return `${field} filter must have at least one value`;
+      }
+    }),
+    validator('array-empty-value', {
```
Comment (jkusa) [inline packages/core/addon/models/bard-request-v2/fragments/filter.js:23]: Going forward I think we can allow this. We went away from checking this in dashboards.

**Pull request merged**